DBugVision/main.py

```diff
@@ -16,38 +18,106 @@ def filter_sort_contours(contours):
              This list may not contain all the original DbugContours in contours, since we filter the given list for
              Better performance.
     """
-    pass
+    potential_bounders = []
+    for contour in contours:
+        (x_offset, y_offset, width, height) = contour.straight_enclosing_rectangle()
+        if not (MIN_BOUND_RECT_AREA < contour.contour_area() < MAX_BOUND_RECT_AREA):
+            logger.debug("Rect denied, min-max rect area with area: " + str(width*height))
+            continue
+
+        ratio = float(height) / float(width)
+        if (MIN_HEIGHT_WIDTH_RATIO > ratio) or (MAX_HEIGHT_WIDTH_RATIO < ratio):
+            logger.debug("Rect denied, height/width ratio")
+            print("R" + str(ratio))
+            continue
+
+        potential_bounders.append(contour)
+
+    potential_bounders.sort(key=lambda cnt: cnt.contour_area(), reverse=True)
+
+    return potential_bounders
 
 
 def init_vision_command():
 
     # The communication channel to the robot
-    robot_com = DBugNetworking()
+    robot_com = DBugNetworking(host=ROBORIO_MDNS,port=ROBORIO_PORT)
+
     # Video stream to get the images from
-    cam = DBugVideoStream(camera_device_index=DBugVideoStream.get_camera_usb_device_index())
+    camera_device_index = DBugVideoStream.get_camera_usb_device_index()
+    camera_device_index = 0 if camera_device_index is None else camera_device_index
+
+    print("Found camera index: " + str(camera_device_index))
+
+    if camera_device_index is None:
+        cam = DBugVideoStream(camera_device_index=camera_device_index)
+    else:
+        cam = DBugVideoStream()
 
     return cam, robot_com
 
 
 def run_vision_command(cam, robot_com):
-    # Capture images non-stop, process them and send the results to the robot
+
+    # Capture frames non-stop, process them and send the results to the robot
     while True:

```

Comment:
We need to find a better way of doing this so that we can control the program properly. So that we don't kill the python to rerun it and make sure we close all the handles and everything. Unless there's no such problem and the the current `service restart` should do.